django/db/models/base.py

```diff
@@ -450,6 +450,14 @@ def __init__(self, db=None):
         # Necessary for correct validation of new instances of objects with explicit (non-auto) PKs.
         # This impacts validation only; it has no effect on the actual save.
         self.adding = True
+        self.cache = {}
+
+    def set_fields_cache(self, field, value):
```

Comment:
Function calls are also quite expensive in Python, so the burden of proof is to demonstrate that these accessor methods add significant value over just directly accessing the attribute.